src/load_dotenv.rs

```diff
@@ -1,43 +1,57 @@
 use crate::common::*;
 
-pub(crate) fn load_dotenv(
+fn load_from_file(
   config: &Config,
   settings: &Settings,
-  working_directory: &Path,
+  path: &Path,
 ) -> RunResult<'static, BTreeMap<String, String>> {
   // `dotenv::from_path_iter` should eventually be un-deprecated, see:
   // https://github.com/dotenv-rs/dotenv/issues/13
   #![allow(deprecated)]
 
+  if settings.dotenv_load.is_none()
+    && config.verbosity.loud()
+    && !std::env::var_os("JUST_SUPPRESS_DOTENV_LOAD_WARNING")
+      .map(|val| val.as_os_str().to_str() == Some("1"))
+      .unwrap_or(false)
+  {
+    eprintln!(
+      "{}",
+      Warning::DotenvLoad.color_display(config.color.stderr())
+    );
+  }
+
+  let iter = dotenv::from_path_iter(&path)?;
+  let mut dotenv = BTreeMap::new();
+  for result in iter {
+    let (key, value) = result?;
+    if env::var_os(&key).is_none() {
+      dotenv.insert(key, value);
+    }
+  }
+  Ok(dotenv)
+}
+
+pub(crate) fn load_dotenv(
+  config: &Config,
+  settings: &Settings,
+  working_directory: &Path,
+) -> RunResult<'static, BTreeMap<String, String>> {
   if !settings.dotenv_load.unwrap_or(true) {
     return Ok(BTreeMap::new());
   }
 
-  for directory in working_directory.ancestors() {
-    let path = directory.join(".env");
+  // look directly for the environment file if specified by path
+  let path = working_directory.join(&config.dotenv_path);
+  if path.is_file() {
+    return load_from_file(config, settings, &path);
+  }
```

Comment:
This should return an error if the user passed `--dotenv-path` and the file cannot be loaded, which we can do if we just unconditionally load the file when the flag is passed:

```suggestion
  if let Some(dotenv_path) = &config.dotenv_path {
    return load_from_file(config, settings, &dotenv_path);
  }
```

I don't think we need to do `working_directory.join(&config.dotenv_path)`, since file reads are always done relative to the current working directory.